File: python-basics/part-3.py
def main():
    try:
        with open("sample.csv", "r") as file:
            word_dict = {}
            for line in file:
                words = line.strip().split(",")
                # Move it to a method and call it, makes it cleaner
                for word in words:
                    if word in word_dict:
                        word_dict[word] += 1
                    else:
                        word_dict[word] = 1
            
            sorted_items_dsc = sorted(word_dict.items(), key = lambda item:item[1], reverse=True)
            print("Top 3 movies: ", sorted_items_dsc[:3])
    except FileNotFoundError:
        print("The file 'sample.csv' was not found.")
    except Exception as e:
        print("An error occurred:", str(e))

if __name__=="__main__":
    main()

# Feedback received:
# Add another except block to handle any other potential exceptions.
# Include doc strings when you define methods for better understanding.
# Global try catch for better safety and resource management.
# OOPs implementation from next time onwards.

File: java-basics/part-3.java
import java.io.File;
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Scanner;

class Solution {
    public static void main(String[] args) {
        try (Scanner sc = new Scanner(new File("items.csv"))) {
            HashMap<String, Integer> freq=new HashMap<>();

            while (sc.hasNextLine()) {
                String line=sc.nextLine();
                String[] items=line.split(",");

                for (String item: items) {
                    item=item.trim().toLowerCase();
                    freq.put(item, freq.getOrDefault(item,0)+1);
                }
            }

            List<Map.Entry<String, Integer>> list=new ArrayList<>(freq.entrySet());
            Collections.sort(list, (a,b) -> b.getValue()-a.getValue());
            
            System.out.println("Top 3 items: ");
            for (int i=0; i<3 && i<list.size(); i++) {
                System.out.println(list.get(i).getKey()+"="+list.get(i).getValue());
            }
        } 
        catch (java.io.FileNotFoundException fnfe) {
            System.out.println("File not found: " + fnfe.getMessage());
        }
        catch (Exception e) {
            System.out.println("An error occurred: " + e.getMessage());
        }
    }
}

// Feedback received:
// try catch for better safety and 
// Global try catch for better safety and resource management.
// Include doc strings when you define methods for better understanding, like below:
/**
 * 
 */
// OOPs implementation from next time onwards.
// src/services -> It will have diff services
// src/utils -> Utility functions
// main.java -> main function
// 